pkg/aggregator/aggregator.go

```diff
@@ -156,6 +156,22 @@ func (s *referenceWalker) Start() {
 
 // FilterSpecByPaths remove unnecessary paths and unused definitions.
 func FilterSpecByPaths(sp *spec.Swagger, keepPathPrefixes []string) {
+	// Walk all references to find all unused definitions. This function
```

Comment:
Shouldn't we remove all unused definitions first and then keep the algorithm as it was? Much simpler.

Also `FilterSpecByPaths` should be called `FilterSpecByPathsAndPrune`.

Reply:
Actually I did this because we should NOT remove unused definitions. When we rename definitions in one release, that is, kind of, backward incompatible change. To make sure we would keep backward compatibility we add definitions with a deprecation description and reference them to the new name. This way all tooling can keep the backward compatibility for a release. These deprecation definitions, however, are unused in the Spec (All paths refer to the new names). If we don't do this here, all of those definitions will be removed from the spec. A good example is https://github.com/kubernetes/kubernetes/pull/51191.